### ISSUE TITLE
tabs: shortcut to rename not configurable, default not functional

### ISSUE THREAD

**Issue by pmattern** [lxde/qterminal#188]
The shortcut to rename tabs cannot be configured, default setting `Alt+Shift+S` doesn't work.

Seen running QTermWidget lxde/qtermwidget@45ef247 and QTerminal lxde/qterminal@ee32e84 on Arch Linux i686 and x86_64, LXQt or KDE.
Shortcut `Alt+Shiftt+S` wasn't assigned globally as it was working as expected if assigned to e. g. `Next Tab` from Preferences - Shortcuts.

**Comment by pmattern** [comment]
In f2404/qterminal@fcf8995d6e changing tab titles by shortcut `Alt+Shift+S` works as expected, so the commit seems GTM.

On a side note I was having a hard time cloning your fork as git kept poking me about credentials.
This happened sporadically when invoking git itself from the command line
```
git clone https://github.com/f2404/qterminal.git
```
but always when I tried to install the commit by a modified version of an Arch Linux AUR package ([qterminal-git](https://aur.archlinux.org/packages/qterminal-git/) here) which is the approach I normally always take to perform tests
```
  -> Cloning qterminal git repo...
Cloning into bare repository '/usr/local/peter/Quellen/qterminal'...
Username for 'https://github.com':
```
If you should happen to have an idea what's going on here I'd be glad to know.

**Comment by f2404** [comment]
After making the shortcut work, I have a problem with configuring it.
All shortcuts that can be configured from the settings are assigned to QActions and stored in the map: `QMap< QString, QAction * > actions;`. Now I've introduced `QShortcut *renameSession`, and it cannot be added into existing map.
What approach should I take to make my shortcut configurable? convert it to `QAction` somehow? or add another map for `QShortcuts` with 1 member? else?

**Comment by f2404** [comment]
The issue has been resolved - please see the PR.

**Comment by f2404** [comment]
This can be closed now.

**Comment by pmattern** [comment]
True.

On a side note commit messages can be tweaked to close issues they refer to. Haven't got a link of the official documentation at hand right now.

**Comment by f2404** [comment]
Hmm, I've specified that the PR closes the issue, isn't this enough?

**Comment by pmattern** [comment]
So far I for one was aware of those messages working in commit messages only and didn't even know they are supposed to work in PRs, too. So I cannot tell you what's the matter here, either.

**Comment by f2404** [comment]
Yep, they are: https://github.com/blog/1506-closing-issues-via-pull-requests
I'm thinking maybe the matter is how the PR was merged - git told me that my branch wasn't merged (I had to use '-D' to delete it), so it might be that github didn't take it as merged, too.

**Comment by jleclanche** [comment]
The `Closed #xxx` has to be part of the commit message.

**Comment by pmattern** [comment]
@f2404
Basically those "special keywords" are indeed working if stated in a Pull Request's body only and not in the corresponding commits. But this seems to work only when the PR gets merged in terms of the GitHub web interface or rebased and no further commits had been added to master in the meantime. The keywords stop working when the commits of the PR get rebased and additional commits had been added to master in the meantime, even when there are no merge conflicts. In that case the PR does not get closed automatically when its commits get included in master. The PR has to be closed manually which in turn does not trigger closing the corresponding issue.
At least this applies to the PR we're discussing here and is consistent with findings in a dummy repo "test-repo" I established in my account yesterday. Feel free to add to the latter if you so wish to look into all this further.
